c7n/resources/rds.py

```diff
@@ -281,6 +282,107 @@ def process(self, dbs):
                 **params)
 
 
+@filters.register('upgrade-available')
+class UpgradeAvailable(Filter):
+
+    schema = type_schema('upgrade-available')
+
+    def _list_versions(self, client, resource):
```

Comment:
this method is returning a single version, so plural name seems odd, really its get engine upgrade, and re params should get the engine instead of the resource.